src/OrchardCore/OrchardCore.Abstractions/Json/Serialization/DateTimeJsonConverter.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace OrchardCore.Json.Serialization;
+public class DateTimeJsonConverter : JsonConverter<DateTime>
+{
+    // ISO 8601 format with milliseconds and time zone.
+    private const string DateTimeFormat = "yyyy-MM-ddTHH:mm:ssZ"; 
+
+    public override DateTime Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        if (reader.TokenType != JsonTokenType.String)
+        {
+            throw new JsonException($"Unexpected token parsing DateTime. Expected a string, got '{reader.TokenType}'.");
+        }
+
+        var stringValue = reader.GetString();
+        if (DateTime.TryParse(stringValue, null, DateTimeStyles.RoundtripKind, out var dateTime))
+        {
+            return dateTime;
+        }
+
+        throw new JsonException($"Unable to convert '{stringValue}' to DateTime.");
+    }
+
+    public override void Write(Utf8JsonWriter writer, DateTime value, JsonSerializerOptions options)
+        => writer.WriteStringValue(value.ToString(DateTimeFormat, CultureInfo.InvariantCulture));
```

Comment:
This is the same as the `"o"` format, I don't think we need the format constant.

Reply:
But if you don't format the constants, you get a format like this

![image](https://github.com/OrchardCMS/OrchardCore/assets/15613121/6867b770-e9e4-44cd-bbd7-1a8175cfa8c8)